src/util/expr_cast.h

```diff
@@ -27,7 +27,7 @@ Author: Nathan Phillips <Nathan.Phillips@diffblue.com>
 /// \tparam T The exprt-derived class to check for
 /// \param base Reference to a generic \ref exprt
 /// \return true if \a base is of type \a T
-template<typename T> bool can_cast_expr(const exprt &base);
+template<typename T> inline bool can_cast_expr(const exprt &base);
```

Comment:
This means missing definitions are flagged at compile time rather than link time.

Reply:
That doesn't look like common practice, I'd put that comment in the code.
Also it looks like `can_cast_expr` should have been expressed as a meta function. Inline is merely a suggestion for the compiler to inline. It might work differently on debug and release modes and different compilers.

Reply:
Not so in this case -- if it's declared inline, then the compiler knows it definitely wants a definition in this unit, rather than generating an extern symbol and waiting for the linker to solve the problem. It's still free to choose whether to actually inline or compile out-of-line and flag it for merging with other units' definitions.